be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -194,6 +209,10 @@ Status VerticalGeneralTabletWriter::flush_columns() {
     DCHECK(_segment_writers[_current_writer_index]);
     RETURN_IF_ERROR(flush_columns(&_segment_writers[_current_writer_index]));
     _current_writer_index = 0;
+
+    if (_async_write_segment) {
```

Comment:
should use `_segment_writer_finalize_token` as the condition? in case `_async_write_segment` is on but open() fails. Just a defensive coding practice.